stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -296,3 +388,66 @@ def is_greater_eq(value, other):
                     # compute how much packaging we can get
                     take = (need // pack_quantity) * pack_quantity
                     need = yield location, location_quantity, take, None, None
+
+    @api.model
+    def _get_requested_comparisons(self, rounding):
```

Comment:
Make a method `_compare_with_tolerance(self, need, quantity, rounding)` implementing the comparisons. You don't need wrappers.
I don't find the case where there is no tolerance.

Reply:
> Make a method `_compare_with_tolerance(self, need, quantity, rounding)` implementing the comparisons. You don't need wrappers.

It's not just wrapper, I use closures in my solution. And why wappers is not good solution?

Reply:
I don't understand the benefit of this approach of coding in odoo. You can't override the inner methods.
A plain compare method would be simple and easy to read.